pkg/channel/distributed/controller/kafkachannel/reconciler.go

```diff
@@ -49,6 +49,9 @@ import (
 
 // Reconciler Implements controller.Reconciler for KafkaChannel Resources
 type Reconciler struct {
+	// Note:  The only purpose this logger field serves is to be available for the configObserver function
+	//        which has a predefined definition that contains no context or logger parameters.  Other reconciler
+	//        logging output should be done via the Logger contained in the given Context.
 	logger               *zap.Logger
```

Comment:
Let's chat about this one - I'd rather not leave this reference here (too tempting for accidental future use) and we can possibly create a new one in the `configObserver()` code?

Reply:
The only reason I didn't create a new logger in the configMapObserver is that the logger put into the Reconciler struct here is the one that comes from the context in the controller.go's NewController() function.  I don't know if there's anything in that particular logger that's important (the traceId, if present, is probably not all that useful in the configMapObserver).  If not, making a new logger in the configMapObserver function (and dropping this field) is certainly easy enough.